### PR TITLE
List federated instances

### LINKED ISSUE #3: Get Federated Instances
Are there plans to just fetch the federated instances with something like https://join-lemmy.org/api/interfaces/FederatedInstances.html#allowed api call, which I think would be the call to make in order to get a list like there is on https://join-lemmy.org/instances?

If the answer is yes, it don't look too hard--I can probably take a stab at it. But I want to double check that this is the intent of the project before I write a bunch of code.

Thanks

### PR DESCRIPTION
**Pull request by thebigG**
Hi there,

Fixes  #3 

This is a very rough prototype, but I wanted to get something that just "works" to get feedback asap.

The commit history is a mess at the moment, this is because I've been trying to be keep up with latest changes and solving merge conflicts. I can always rebase/force push later once this is more polished, but for now just posting to get feedback.

Here's a screenshot in case you're in a hurry and don't have time to build :)
lemoa_instances

Things I'm thinking about improving(adding):

- Make each row selectable/clickable. As a user I feel it is slightly inconvenient to have to reach with my mouse all the way to the right to switch instances by clicking on the Choose button. I feel in a similar way about the Communities view.
- Start thinking about how to integrate with all of the auth stuff, which I understand still is WIP?
- Would be nice to get images for each instances, but not sure if they're exposing that already since this is using the very new 0.18.0 version of the lemmy API.
- Add a filter search bar at the very top of the view so we don't have to scroll to infinity to find the instance we're looking for :)

Please let me know if I'm missing anything important--or any other piece of feedback.

Thanks

### CODE REVIEW

File: src/api/instances.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        auth: settings::get_current_account().jwt,
+    };
+     let instances: std::result::Result<GetFederatedInstancesResponse, reqwest::Error>  = super::get("/federated_instances", &params);
+
+    
+    match instances
+    {
+        Ok(instances) =>
+        {
+               instances.federated_instances
```
Comment (FireMasterK) [inline src/api/instances.rs:18]: This also includes other federated software instances like mastodon, I think this should be filtered.

**Comment by Bnyro** [comment]
1. Yes, I'd like to do that too, but unfortunately I don't yet know how we can make normal components click able with Relm4 :(
2. All common auth features are already available, you can find them after logging in with the three dot menu at the top right.
3. The images would be nice too have, but it's not really important for now.
4. This would be cool, and should be quite straight forward to implement. You can just add a Gtk entry and filter the instances when the text got changed, a bit similar to how it's done in the communities list.

In general I feel like it would be more convenient for users if the list of instances was part of the page where you choose an instance (3 dot menu on the top right -> Choose instance), because we now have two different places to choose an instance which might be a bit confusing for new users.

**Comment by Bnyro** [comment]
Concerning 1.: I figured that out now, see https://github.com/lemmygtk/lemoa/commit/380e95235d2c7777ecb3aa7bc6d48a91b2767c56 for an example of how to implement it :)

**Comment by Bnyro** [comment]
Made some adjustments to show it to new users on the first app startup. Thank you!

**Comment by Bnyro** [comment]
Ah damn, the commit log is broken now because I forgot to run `git add`.
The rest of the changes from the PR are located in https://github.com/lemmygtk/lemoa/commit/1e4ccacfe80f4a4fcef43b038913160233e1697a.

**Pull request merged**